nginx-controller/controller/utils.go

```diff
@@ -163,10 +163,21 @@ type StoreToIngressLister struct {
 	cache.Store
 }
 
+// GetByKey calls Store.GetByKey and returns a copy of the ingress so it is
+// safe to modify.
+func (s *StoreToIngressLister) GetByKey(key string) (ing *extensions.Ingress, exists bool, err error) {
```

Comment:
perhaps it makes sense to use a different name for GetByKey? Consider that in controller.go we call GetByKey of the other Listers, which will continue to return an unsafe to modify resource